source/blender/blenkernel/intern/sca.c

```diff
@@ -1188,6 +1190,23 @@ void BKE_sca_actuators_id_loop(ListBase *actlist, SCAActuatorIDFunc func, void *
   }
 }
 
+void BKE_sca_components_id_loop(ListBase *complist, SCAComponentIDFunc func, void *userdata)
+{
+  PythonComponent *comp;
+
+  for (comp = complist->first; comp; comp = comp->next) {
+      ListBase *properties = &comp->properties;
+      PythonComponentProperty *prop;
+
+      for (prop = properties->first; prop; prop = prop->next) {
+#define PT_DEF(name, lower, upper) \
+          func(comp, (ID **)&prop->lower, userdata, IDWALK_CB_NOP);
```

Comment:
I'm not sure IDWALK_CB_NOP should be used in all cases

Reply:
Changed it along with a few similar parts regarding actuators. This should fix the erraneous behaviour which was mentioned above. Thanks for pointing this out!